### DISCUSSION TITLE
A hook called `use` without any suffix causes error in react compiler

### DISCUSSION THREAD

**Discussion by stephanoskomnenos** [reactwg/react-compiler#47] [Something went wrong]
I got `Uncaught Error: Rendered fewer hooks than expected.`. Without compiler it works file.

**Comment by stephanoskomnenos** [comment]
If I change its name to `use1`, it works fine in react compiler.

**Reply by hameda169** [reply]
Make sure that you don't use any hook inside of a condition or loop block. Also you shouldn't have any return statement before calling a hook.

**Reply by stephanoskomnenos** [reply]
> Make sure that you don't use any hook inside of a condition or loop block. Also you shouldn't have any return statement before calling a hook.

Certainly. I made a reproduction codesandbox.

**Reply by hameda169** [reply]
I see. Maybe it's a conflict with react's built-in `use` hook that introduced in React 19 release.

**Reply by stephanoskomnenos** [reply]
> I see. Maybe it's a conflict with react's built-in `use` hook that introduced in React 19 release.

I think it's not the reason. I didn't import that `use` hook and this bug can be reproduced in react 18 with react compiler too.

**Reply by NobodyXu** [reply]
> I think it's not the reason. I didn't import that `use` hook and this bug can be reproduced in react 18 with react compiler too.

Maybe the react compiler isn't smart enough to tell, and always treats `use` as the react 19 built-in hook?

**Comment by stephanoskomnenos** [comment]
Here is the minimal reproduction: https://codesandbox.io/p/devbox/epic-jepsen-3vnscm?workspaceId=ws_6kEPchLcYbhhULqCZ4Y2rh

**Comment by stephanoskomnenos** [comment]
solved in latest RC version

**Reply by stephanoskomnenos** [reply]
> solved in latest RC version

Reopened it. Not fully solved.
